### ISSUE TITLE
Add image/thumbnail view to directories

### ISSUE THREAD

**Issue by gyscos** [svenstaro/miniserve#139]
In a directory of images, it could be nice to have a way to preview the images.
One naive solution would be to directly embed the image in the page; a maybe better way would be to auto-generate thumbnails on the fly to save on bandwidth (and maybe cache them in a temporary directory), but that may be trickier.

**Comment by svenstaro** [comment]
Yeah, that would be really cool. I think given that we're using rust, we can easily generate them on the server on the fly in parallel.

**Comment by HerHde** [comment]
And maybe add a gallery view to cycle through the images with the arrow keys instead of going back and clicking the next one.
Even if it's a bit off-topic...

**Comment by WoMspace** [comment]
bump
(apologies for necro)

**Comment by svenstaro** [comment]
@WoMspace are you interested in taking a stab at this?

**Comment by WoMspace** [comment]
gosh you know it seems the perfect kind of project for me to brush up on rust, but I'm absolutely booked full.
Assuming this issue is still open in a few months and I remember to come back to it, I do think I'll take a stab at it

**Comment by svenstaro** [comment]
Alright as I'm somewhat unlikely to work on this myself anytime soon I think there's a good chance you can just work on this when to in feel like it. I still think it's a cool feature if implemented neatly.

**Comment by intergalacticmonkey** [comment]
This would be really great. There's no need to reinvent the wheel, either: https://askubuntu.com/a/1368911, https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html 

Thanks to this awesome spec, all you have to do is parse the thumbnailer entries in `/usr/share/thumbnailers` and `~/.local/share/thumbnailers` and execute the specified programs for each file of each mime type that you want to display thumbnails for (and store them in a cache directory). GNOME's built-in thumbnailers, and most installable thumbnailers such as `ffmpegthumbnailer` use this spec and drop files in the "thumbnailers" directories.

**Comment by Atreyagaurav** [comment]
Gallery view with checkbox for selecting and downloading would make the image sharing so much smoother. This could even become a preferable app to let friends download the pictures you took from your phone as well (although a filter to only share files within a datetime limit might be required for safety reason).

Is anyone going to take this? I'd offer to, but most of this is trying to make the front end look good, that goes out of my expertise.
